File: src/main/java/com/nugurang/graphql/resolver/PositionResolver.java
package com.nugurang.graphql.resolver;

import com.nugurang.dao.PositionDao;
import com.nugurang.dto.ImageDto;
import com.nugurang.dto.PositionDto;
import graphql.kickstart.tools.GraphQLResolver;
import java.util.Optional;
import org.springframework.stereotype.Service;

@Service
public class PositionResolver implements GraphQLResolver<PositionDto> {
    private final PositionDao positionDao;

    public Optional<ImageDto> image(PositionDto positionDto) {
        return Optional.ofNullable(positionDao.findById(positionDto.getId()).get().getImage()).map(entity -> entity.toDto());
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public PositionResolver(final PositionDao positionDao) {
        this.positionDao = positionDao;
    }
    //</editor-fold>
}

File: src/main/kotlin/com/nugurang/graphql/resolver/PositionResolver.kt
package com.nugurang.graphql.resolver

import com.nugurang.dao.PositionDao
import com.nugurang.dto.ImageDto
import com.nugurang.dto.PositionDto
import com.nugurang.mapper.ImageMapper
import graphql.kickstart.tools.GraphQLResolver
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
class PositionResolver(
    private val positionDao: PositionDao,
    private val imageMapper: ImageMapper
) : GraphQLResolver<PositionDto> {
    // TODO: return nullable or non-nullable with error?
    fun image(positionDto: PositionDto): ImageDto? {
        return positionDao.findByIdOrNull(positionDto.id)?.image?.let(imageMapper::toDto)
    }
}